### PR TITLE
Api support for 800/1000/1200 for Interconnects

### PR DESCRIPTION
**Pull request by AsisBagga**
**Description**
Api support for 800/1000/1200 for Interconnects

**Issues Resolved**
N/A

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: examples/logical_interconnect.rb [hunk 1]
```diff
@@ -15,34 +15,17 @@
 # NOTE 3: The api_version client should be 300 or greater if you run the examples using Scopes
 
 my_client = {
-  url: ENV['ONEVIEWSDK_URL'],
-  user: ENV['ONEVIEWSDK_USER'],
-  password: ENV['ONEVIEWSDK_PASSWORD']
+  url: 'https://10.170.16.33',
```
Comment (soodpr) [inline examples/logical_interconnect.rb]: don't replace these env variables with actual values

File: libraries/resource_providers/api500/synergy/interconnect_provider.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+# (c) Copyright 2020 Hewlett Packard Enterprise Development LP
```
Comment (soodpr) [inline libraries/resource_providers/api500/synergy/interconnect_provider.rb]: we don't need to update the copyright for files which have no updates

File: libraries/resource_providers/api600/synergy/interconnect_provider.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# (c) Copyright 2018 Hewlett Packard Enterprise Development LP
+# (c) Copyright 2020 Hewlett Packard Enterprise Development LP
```
Comment (soodpr) [inline libraries/resource_providers/api600/synergy/interconnect_provider.rb]: same as above

**Review by soodpr — CHANGES_REQUESTED** [review]
Just noticed that you have updated Logical interconnects example file but your PR supports Interconnects.

File: endpoints-support.md [hunk 4]
```diff
@@ -69,6 +69,12 @@
...
 |<sub>/rest/fcoe-networks/{id}</sub>                                                      | PUT      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
 |<sub>/rest/fcoe-networks/{id}</sub>                                                      | DELETE   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+
+|     **Interconnects**                                                                                                                               |
+|<sub>/rest/interconnects/{id}</sub>                                                        | PATCH    | :white_check_mark:  | :white_check_mark:   | :white_check_mark:   | :white_check_mark:  |:white_check_mark:   | :white_check_mark: | :white_check_mark:  |
```
Comment (soodpr) [inline endpoints-support.md]: have you added all the endpoints supportedfor Interconnects? interconnect configuration is missing...verify all others as well.

**Review by soodpr — CHANGES_REQUESTED** [review]
Endpoint-support.md file is not formatted correctly.

File: endpoints-support.md [hunk 5]
```diff
@@ -69,18 +69,23 @@
...
-|<sub>/rest/logical-enclosures/{id}/support-dumps</sub>                                   | POST     | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:  | :white_check_mark:  | :white_check_mark:  | :white_check_mark:  |
-|<sub>/rest/logical-enclosures/{id}/updateFromGroup</sub>                                 | PUT      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:  | :white_check_mark:  | :white_check_mark:  | :white_check_mark:  |
+|   **Interconnects**                                                                                                                           |
+|<sub>/rest/interconnects/{id}</sub>                                                    | PATCH    | :white_check_mark:  | :white_check_mark:   | :white_check_mark:   | :white_check_mark:  |:white_check_mark:   | :white_check_mark: | :white_check_mark:  |
+|<sub>/rest/interconnects/{id}/configuration</sub>                                      | PUT    | :white_check_mark:  | :white_check_mark:   | :white_check_mark:  | :white_check_mark:  |:white_check_mark:   | :white_check_mark: | :white_check_mark:  |
```
Comment (soodpr) [inline endpoints-support.md:74]: This endpoint supported from API 500 onwards.please mark it correctly here

File: examples/interconnect.rb [hunk 6]
```diff
@@ -53,7 +53,7 @@
   action :reset
 end
 
-# Only for api500
+# for Api grater than Api500
```
Comment (soodpr) [inline examples/interconnect.rb]: typo error for greater

**Review by soodpr — APPROVED** [review]

File: examples/logical_interconnect.rb (hunk 1 above)
Comment (AsisBagga) [reply examples/logical_interconnect.rb]: @soodpr Done

File: libraries/resource_providers/api500/synergy/interconnect_provider.rb (hunk 2 above)
Comment (AsisBagga) [reply libraries/resource_providers/api500/synergy/interconnect_provider.rb]: @soodpr Done

File: libraries/resource_providers/api600/synergy/interconnect_provider.rb (hunk 3 above)
Comment (AsisBagga) [reply libraries/resource_providers/api600/synergy/interconnect_provider.rb]: @soodpr Done

File: endpoints-support.md (hunk 4 above)
Comment (AsisBagga) [reply endpoints-support.md]: @soodpr Done

File: endpoints-support.md (hunk 5 above)
Comment (AsisBagga) [reply endpoints-support.md:74]: @soodpr Done

File: examples/interconnect.rb (hunk 6 above)
Comment (AsisBagga) [reply examples/interconnect.rb]: @soodpr Done

**Review by AnushaY1916 — APPROVED** [review]

**Pull request merged**